File: services/core/src/main/java/me/unbrdn/core/interview/application/port/in/PauseInterviewUseCase.java
package me.unbrdn.core.interview.application.port.in;

import java.util.UUID;

/** Use Case for pausing an interview */
public interface PauseInterviewUseCase {
    PauseInterviewResult execute(PauseInterviewCommand command);

    record PauseInterviewCommand(UUID interviewId) {}

    record PauseInterviewResult(String interviewId, String status, String pausedAt) {}
}

File: services/gateways/bff/src/modules/interview/usecases/pause-interview.usecase.ts
import { Injectable } from "@nestjs/common";
import { InterviewGrpcService } from "../../../infra/grpc/services/interview-grpc.service";

export class PauseInterviewCommand {
    constructor(public readonly interviewId: string) {}
}

@Injectable()
export class PauseInterviewUseCase {
    constructor(private readonly interviewGrpcService: InterviewGrpcService) {}

    async execute(command: PauseInterviewCommand): Promise<{
        interviewId: string;
        status: string;
        pausedAt: string;
    }> {
        return await this.interviewGrpcService.pauseInterview(command.interviewId);
    }
}
